### PR TITLE
Fix and refactor API changelogs

### PR DESCRIPTION
**Pull request by lzdanski**
Adds changelogs for the following PR's:

- #3846 
- #3869 
- #3896 
- #3990 

Preview links for review: 

- [Platform API changelog](https://astronomer-preview-5b4e541a-8a80-4c9c-a600-6181f5694af7.docs.buildwithfern.com/platform-api-reference/changelog)
- [IAM API changelog](https://astronomer-preview-5b4e541a-8a80-4c9c-a600-6181f5694af7.docs.buildwithfern.com/iam-api-reference/changelog)

### CODE REVIEW

File: api/fern/apis/platform/changelog/2024-07-10.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## Summary
+
+This update introduces new endpoints for retrieving the health statuses of different Deployment resources. See [Astro documentation](https://docs.astronomer.io/astro/deploy-code) for more information.
```
Comment (astro-walker) [inline api/fern/apis/platform/changelog/2024-07-10.md:3]: This is not correct, these statuses refer to the cluster which contains the deployments, not the deployments themselves. Relevant doc links are
https://www.astronomer.io/docs/astro/create-dedicated-cluster
https://www.astronomer.io/docs/api/platform-api-reference/cluster/list-clusters#response.body.clusters.healthStatus

Comment (lzdanski) [inline api/fern/apis/platform/changelog/2024-07-10.md:3]:
```suggestion
This update introduces new endpoints for retrieving the health statuses of different clusters. See [Astro dedicated cluster documentation](https://www.astronomer.io/docs/astro/create-dedicated-cluster) for more information.
```

File: api/fern/apis/iam/changelog/2024-07-31.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+## Summary
+
+This update introduces new endpoints for managing IP access lists. See [Astro documentation](https://docs.astronomer.io/astro/deploy-code) for more information.
```
Comment (astro-walker) [inline api/fern/apis/iam/changelog/2024-07-31.md:3]: The docs for this feature are here: https://www.astronomer.io/docs/astro/ip-access-list

File: api/fern/apis/platform/changelog/2024-07-31.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+## Summary
+
+This update introduces new endpoints for managing IP access lists. See [Astro documentation](https://docs.astronomer.io/astro/deploy-code) for more information.
```
Comment (astro-walker) [inline api/fern/apis/platform/changelog/2024-07-31.md:3]: See comment above, there is a more specific doc link

File: api/fern/apis/iam/changelog/2024-07-31.md (hunk 2 above)
Comment (lzdanski) [inline api/fern/apis/iam/changelog/2024-07-31.md:3]:
```suggestion
This update introduces new endpoints for managing IP access lists. See [Astro IP access list documentation](https://www.astronomer.io/docs/astro/ip-access-list) for more information.
```

File: api/fern/apis/platform/changelog/2024-07-31.md (hunk 3 above)
Comment (aliotta) [reply api/fern/apis/platform/changelog/2024-07-31.md:3]: https://www.astronomer.io/docs/astro/ip-access-list

Comment (lzdanski) [inline api/fern/apis/platform/changelog/2024-07-31.md:3]:
```suggestion
This update introduces new endpoints for managing IP access lists. See [Astro documentation](https://www.astronomer.io/docs/astro/ip-access-list) for more information.
```

**Review by aliotta — APPROVED** [review]

**Pull request merged**